manager-bundle/src/Command/ContaoSetupCommand.php

```diff
@@ -56,6 +59,25 @@ protected function configure(): void
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
+        $io = new SymfonyStyle($input, $output);
+
+        // Auto-generate a kernel secret if none was set
+        if (empty($this->kernelSecret) || 'ThisTokenIsNotSoSecretChangeIt' === $this->kernelSecret) {
+            $filesystem = new Filesystem();
+
+            $dotenv = new DotenvDumper(Path::join($this->projectDir, '.env.local'), $filesystem);
+            $dotenv->setParameter('APP_SECRET', bin2hex(random_bytes(32)));
+            $dotenv->dump();
+
+            $io->info('An APP_SECRET was generated and written to your .env.local file.');
+
+            if (!$filesystem->exists($envPath = Path::join($this->projectDir, '.env'))) {
```

Comment:
Why this? Is it technically necessary to have both an `.env` and `.env.local` file? Because if not, we should not generate the `.env` file.

Reply:
The `.env.local` will only be loaded by Symfony's Dotenv component, if an `.env` file is present.

Reply:
Why not use a `.env.local.php` file? It does not require an `.env` file to be present and it is the fastest way of using .env files.

Reply:
The `.env.local.php` is created by running `composer dump-env prod`, you should not generate it yourself. For the `APP_SECRET` to be defined in the `.env.local.php`, it needs to be defined in the `.env.local` first.